src/metax_api/utils/reference_data_service.py

```diff
@@ -59,12 +59,18 @@ def _fetch_reference_data(cls, settings):
         esclient = Elasticsearch(settings['HOSTS'], **connection_params)
         indicesclient = IndicesClient(esclient)
         reference_data = {}
-
         for index_name, index_info in indicesclient.get_mapping().items():
             reference_data[index_name] = {}
             for type_name in index_info['mappings'].keys():
+                reference_data[index_name][type_name] = []
                 all_rows = scan(esclient, query={'query': {'match_all': {}}}, index=index_name, doc_type=type_name)
-                reference_data[index_name][type_name] = [ row['_source']['uri'] for row in all_rows if row['_source']['uri'] ]
+                for row in all_rows:
+                    entry = {}
+                    if row['_source']['uri']:
```

Comment:
Isn't uri always present? At least the code still runs if the 'if' is removed

Reply:
I guess it's a matter of principle - do we always and forever trust there is a uri for any reference data. There may be human errors etc. which may justify this piece of code. What do you think?

Reply:
If it's just a precaution for errors in the ref data, consider logging an error when its missing so it can be spotted and fixed.